mantidproject/mantid
# Add file back option to ConvertToMD, CreateMD and AccumulateMD

ConvertToMD is responsible for creating the MDEventWorkspace (it is called as a child alg by CreateMD and AccumulateMD). It now sets the fileback option on the box controller of this workspace and saves the workspace.

Release notes (561a10d) and user docs for the algs (0ca00ed) were updated.

**To test:**
- Get the data to test with from //olympic/babylon5/Scratch/Matthew Jones/Fe_data.
- In the script below, replace the path with the correct path to the data, then run it.
- Compare the file-backed and normal output workspaces (details in the dock and could compare visually in Slice Viewer), they should be the same.

``` python
import numpy as np

# Create arrays of run numbers and corresponding values of psi
run_numbers = range(15052, 15062)
psi_array = np.arange(0.0, 20.0, 2)

input_runs = ['/path/to/Fe_data/map' + str(run_number) + '_ei400.nxspe' for run_number in run_numbers]

md_ws = CreateMD(input_runs, Emode='Direct', Alatt=[2.87, 2.87, 2.87], Angdeg=[90, 90, 90], u=[1, 0, 0, ], v=[0, 1, 0], Psi=psi_array, EFix=400.0, InPlace=True)
md_ws_fileback = CreateMD(input_runs, Emode='Direct', Alatt=[2.87, 2.87, 2.87], Angdeg=[90, 90, 90], u=[1, 0, 0, ], v=[0, 1, 0], Psi=psi_array, EFix=400.0, InPlace=True, FileBackEnd=True, Filename="test_fb_createmd.nxs")
```

Fixes #14949.



---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
